src/Role.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Tuf;
+
+/**
+ * Class that represents a TUF role.
+ */
+class Role
+{
+
+    /**
+     * The role name.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * The role threshold.
+     *
+     * @var int
+     */
+    protected $threshold;
+
+    /**
+     * The key IDs.
+     *
+     * @var array
+     */
+    protected $keyIds;
+
+
+    /**
+     * Role constructor.
+     *
+     * @param string $name
+     *   The name of the role.
+     * @param int $threshold
+     *   The role threshold.
+     * @param array $keyIds
+     *   The key IDs.
+     */
+    private function __construct(string $name, int $threshold, array $keyIds)
+    {
+        $this->name = $name;
+        $this->threshold = $threshold;
+        $this->keyIds = $keyIds;
+    }
+
+    /**
+     * Creates a role object from TUF metadata.
+     *
+     * @param \ArrayObject $roleInfo
+     *   The role information from TUF metadata.
+     * @param string $name
+     *   The name of the role.
+     *
+     * @return static
+     *
+     * @see @see https://github.com/theupdateframework/specification/blob/v1.0.9/tuf-spec.md#4-document-formats
+     */
+    public static function createFromMetadata(\ArrayObject $roleInfo, string $name): self
+    {
+        return new static(
```

Comment:
Do we want to add some validation?

Reply:
The metadata will always be validate when created so unless some creates their own ArrayObject instance I don't see how it could have bad data.

Reply:
I mean...it's pretty easy to create an ArrayObject instance and fill it with bad data. So I'm still +1 for validation, but I'm okay deferring that to a follow-up.

Reply:
Added validation